src/router.ts

```diff
@@ -1,49 +1,145 @@
+import storage from './storage';
 import { $, $$ } from './utils';
 
-interface IRouter {
-  path: string;
-  component: Element;
-}
+const $nav = $('.nav');
+const $loginButton = $('.login-button');
+const $selectBoxWrapper = $('.select-box-wrapper');
+const $userInfoButton = $('.user-info-button');
+const $selectBox = $('.select-box');
+const $purchaseTab = $('purchase-tab');
+const $header = $('.header');
+const $signInPage = $('sign-in');
+const $signUpPage = $('sign-up');
+const $editProfilePage = $('edit-profile');
 
-const nav = document.querySelector('.nav');
 const baseURL = '/javascript-vendingmachine';
 
-nav.addEventListener('click', (e) => {
-  if ((e.target as HTMLButtonElement).type === undefined) return;
+export type Path =
+  | '/javascript-vendingmachine/'
+  | '/javascript-vendingmachine/charge'
+  | '/javascript-vendingmachine/purchase'
+  | '/javascript-vendingmachine/signin'
+  | '/javascript-vendingmachine/logout'
+  | '/javascript-vendingmachine/signup'
+  | '/javascript-vendingmachine/editprofile';
 
-  const route = (e.target as HTMLButtonElement).getAttribute('route');
-  historyRouterPush(route);
-});
+interface IRouter {
+  path: Path;
+  component: Element;
+}
 
-const historyRouterPush = (pathname: string) => {
+export const historyRouterPush = (pathname: Path) => {
   if (pathname === window.location.pathname) return;
-
+  if (pathname === '/javascript-vendingmachine/logout') pathname = '/javascript-vendingmachine/';
   history.pushState({ pathname }, '', pathname);
   render(pathname);
 };
 
-const render = (path: string) => {
+const accessTokenTabRender = (tabName) => {
+  $(tabName).classList.remove('hidden');
+  $nav.classList.remove('hidden');
+  $loginButton.classList.add('hidden');
+  $selectBoxWrapper.classList.remove('hidden');
+  $selectBox.classList.add('hidden');
+  $userInfoButton.classList.remove('hidden');
+  $userInfoButton.textContent = storage.getUserInfo().userName.slice(0, 1);
+};
+
+const noAccessTokenTabRender = () => {
+  $nav.classList.add('hidden');
+  $loginButton.classList.remove('hidden');
+  $purchaseTab.classList.remove('hidden');
+};
+
+const authCommonRender = () => {
+  $selectBoxWrapper.classList.add('hidden');
+  $header.classList.add('hidden');
+  $nav.classList.add('hidden');
+};
+
+const render = (path: Path) => {
   $$('.focus-button').forEach((button) => button.classList.remove('focus-button'));
-  $(`[route='${path}']`, nav)?.classList.add('focus-button');
+  $(`[route='${path}']`, $nav)?.classList.add('focus-button');
 
-  const currentComponent = routers.find((route) => route.path === path)?.component ?? $('product-management');
-  const prevRoute = routers.filter((route) => route.path !== path);
+  switch (path) {
+    case `${baseURL}/`:
+      $header.classList.remove('hidden');
+      if (storage.getAccessToken()) {
```

Comment:
```suggestion
  const isLoggedIn = storage.getAccessToken() !== '';
  switch (path) {
    case `${baseURL}/`:
      $header.classList.remove('hidden');
      if (isLoggedIn) {
```
getAcessToken으로도 충분하다고 생각되지만 LoggedIn이 더 직관적인거같아요!